core/src/main/scripts/importer/validateData.py

```diff
@@ -3229,87 +3260,238 @@ def parse_q_value(self, value, column_number):
+            self.logger.error('Feature id contains one or more illegal characters',
+                                extra={'line_number': self.line_number,
+                                        'cause': 'id was`'+feature_id+'` and only alpha-numeric, _ and - are allowed.'})
         else:
-            # Check if this is the second gene set data file
-            if prior_validated_geneset_ids is not None:
-                # Check if gene set is in the first gene set data file
-                if geneset_id not in prior_validated_geneset_ids:
-                    self.logger.error('Gene sets in cannot be found in other gene set file',
+            # Check if this is the second treatment data file
+            if self.get_prior_validated_feature_ids() is not None:
+                # Check if treatment is in the first treatment data file
+                if feature_id not in self.get_prior_validated_feature_ids():
+                    self.logger.error('Feature id cannot be found in other data file',
                                       extra={'line_number': self.line_number,
-                                             'cause': geneset_id})
-            # Add gene set to list of gene sets of current gene set data file
-            self.geneset_ids.append(geneset_id)
-        return geneset_id
+                                             'cause': feature_id})
+            # Add treatment to list of treatments of current treatment data file
+            self.feature_ids.append(feature_id)
+        return feature_id
 
     def onComplete(self):
 
-        def checkConsistencyScoresPvalue(self):
-            """This function validates whether the gene sets in the scores and p-value file are the same"""
+        def checkConsistencyFeatures(self):
+            """This function validates whether the treatments in the treatment response files (IC50, EC50, 
+            GI50, AUC, ...) are the same"""
 
-            global prior_validated_geneset_ids
-
-            # If the prior_validated_geneset_ids is not filled yet, fill it with the first file.
-            if prior_validated_geneset_ids is None:
-                prior_validated_geneset_ids = self.geneset_ids
+            # If the prior_validated_features_ids is not filled yet, fill it with the first file.
+            if self.get_prior_validated_feature_ids() is None:
+                ids = self.feature_ids
+                self.set_prior_validated_feature_ids(ids)
             else:
-                # Check if gene set ids are the same
-                if not prior_validated_geneset_ids == self.geneset_ids:
-                    self.logger.error(
-                        'Gene sets column in score and p-value file are not equal')
+                # Check if feature ids are the same
+                if not self.get_prior_validated_feature_ids() == self.feature_ids:
+                    log_info = self.get_message_features_do_not_match()
```

Comment:
The abstract method you call here really needs a docstring, to tell subclassers what they should make it return. It seems to be dictionary containing only the one specific key 'cause', that maps to [the violating string found in the data file](https://github.com/cBioPortal/cbioportal/blob/v2.1.0/core/src/main/scripts/importer/cbioportal_common.py#L296)?

Reply:
Docstring has been added.

Reply:
The field that you documented as ‘text that details cause of the error’ is being logged as if it were a string found in the input file. But if you feel that that won't be an issue for the user experience, then it shouldn't block this pull request.

Reply:
I have removed the (optional) 'cause' element. This part of the code compares the lists of identifiers between two files and returns an error when not identical. Since this can be an issue of order or contents. Since it will be rather complicated to find the error I decided to remove the cause component.

Reply:
Sure, should be fine; I said that it should block this pull request, so I'll resolve this thread.